### PR TITLE
feat(node-steam-openid): add types

### PR DESCRIPTION
**Pull request by joshuajeschek**
https://github.com/LeeviHalme/node-steam-openid
https://github.com/LeeviHalme/node-steam-openid/blob/master/index.js

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/node-steam-openid/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Type definitions for node-steam-openid 1.0
+// Project: https://github.com/LeeviHalme/node-steam-openid (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/node-steam-openid/index.d.ts:2]: Please remove left over comments from template (a.k.a. instructions)

File: types/node-steam-openid/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * ```
+     * const steam = new SteamAuth({
+     *   realm: "http://localhost:5000",
+     *   returnUrl: "http://localhost:5000/auth/steam/authenticate",
+     *   apiKey: "XXXXXXXXXXXXXXXXXXXXXXXXXX"
+     * });
+     * ```
+     */
+    constructor({ realm, returnUrl, apiKey }: SteamAuthOptions);
```
Comment (peterblazejewicz) [inline types/node-steam-openid/index.d.ts:26]:
```suggestion
    constructor(steamAuthOptions: SteamAuthOptions);
```

File: types/node-steam-openid/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Gets the redirect URL to Steam.
```
Comment (peterblazejewicz) [inline types/node-steam-openid/index.d.ts:29]:
```suggestion
     * Gets the redirect URL to Steam.
     * @async
```

File: types/node-steam-openid/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    getRedirectUrl(): Promise<string>;
+
+    /**
+     * Authenticates the user with oAuth.
```
Comment (peterblazejewicz) [inline types/node-steam-openid/index.d.ts:34]:
```suggestion
     * Authenticates the user with oAuth.
     * @async
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@joshuajeschek thx!

**Comment by joshuajeschek** [comment]
ready to merge! :)

**Pull request merged**